org.opent2t.sample.thermostat.superpopular/com.smartthings.thermostat/js/thingTranslator.js

```diff
@@ -238,9 +240,184 @@ function validateResourceGet(resourceId) {
+        var providerUnit = getUnitSafe(attributes);
+        var min = getMinHeatingSetpoint(attributes);
+        var max = getMaxCoolingSetpoint(attributes);
+        if (unit === 'c' && providerUnit === 'f') {
+            value = celsiusToFahrenheit(value);
+        } else if (unit === 'f' && providerUnit === 'c') {
+            value = fahrenheitToCelsius(value);
+        }
+        if (value >= min && value <= max) {
+            return unit;
+        }
+        throw new OpenT2TError(440, "Invalid temperature (" + value + ") for unit (" + unit + "[" + min + ", " + max + "])");
+    } else {
+        var providerUnit = getUnitSafe(attributes);
+        var min = getMinHeatingSetpoint(attributes);
+        var max = getMaxCoolingSetpoint(attributes);
+        if (value >= min && value <= max) {
+            return providerUnit;
+        }
+        if (providerUnit === 'c') {
+            min = celsiusToFahrenheit(min);
+            max = celsiusToFahrenheit(max);
+            if (value >= min && value <= max) {
+                return 'f';
+            }
+        } else if (providerUnit === 'f') {
+            min = fahrenheitToCelsius(min);
+            max = fahrenheitToCelsius(max);
+            if (value >= min && value <= max) {
+                return 'c';
+            }
+        }
+        throw new OpenT2TError(440, "Temperature outside supported range (" + value + ")");
+    }
+}
+
+/**
+ * Given a target temperature, compute a target 
+ * low and target high centerd around it.
+ * @param {*} resourceSchema 
+ * @param {*} stateReader 
+ */
+function getTargetTemperatureRange(resourceSchema, attributes) {
+    var result = {};
+
+    var unit = getValidatedUnit(resourceSchema, attributes);
+    var providerUnit = getUnitSafe(attributes);
+
+    var value = resourceSchema.temperature;
+    if (unit === 'c' && providerUnit === 'f') {
+        value = celsiusToFahrenheit(value);
+    } else if (unit === 'f' && providerUnit === 'c') {
+        value = fahrenheitToCelsius(value);
+    }
+
+    // cooling = temperatureHigh
+    var minTemperatureHigh = getMinCoolingSetpoint(attributes);
+    var maxTemperatureHigh = getMaxCoolingSetpoint(attributes);
+
+    // heating = temperatureLow
+    var minTemperatureLow = getMinHeatingSetpoint(attributes);
+    var maxTemperatureLow = getMaxHeatingSetpoint(attributes);
+
+    var targetTemperatureHigh = getIntSafe(attributes.coolingSetpoint);
+    var targetTemperatureLow = getIntSafe(attributes.heatingSetpoint);
+
+    var range = targetTemperatureHigh - targetTemperatureLow;
+    var halfRange = range / 2;
+    var newTargetTemperatureHigh = value + halfRange;
+    var newTargetTemperatureLow = value - halfRange;
+
+    // If either value is outside min/max range, adjust accordingly
+    if (newTargetTemperatureLow < minTemperatureLow) {
+        newTargetTemperatureLow = minTemperatureLow;
+        newTargetTemperatureHigh = minTemperatureLow + range;
+    }
```

Comment:
did you miss an 'else' here?

Reply:
Keeping it consistent with the other translators.  if/else could be more efficient.  Will add to clean up.